scripts/dispatch_table_helper_generator.py

```diff
@@ -168,17 +173,21 @@ def AddCommandToDispatchList(self, name, handle_type, protect, cmdinfo):
             if "VK_VERSION" not in self.featureName and self.extension_type == 'device':
                 self.device_extension_list.append(name)
                 # Build up stub function
-                return_type = ''
                 decl = self.makeCDecls(cmdinfo.elem)[1]
-                if 'typedef VkResult' in decl:
-                    return_type = 'return VK_SUCCESS;'
+                return_type = cmdinfo.elem.find('proto/type').text
+                return_statement = ""  # default type is void, so no return type
```

Comment:
Now that you have the undecorated return type you can just use `[]` to get the correct return value from `return_type_table` and you don't have to loop.

If you still want to output a better error message, you can try/except to catch the KeyError if a return_type_table is missing an entry:
```
try:
    return_statement = 'return ' + return_type_table[return_type] + ';'
except KeyErrror:
    raise AssertionError(...)
```
I promise I am not trying to code-golf your PR, using `[]` makes the code easier to understand

Reply:
No this is good feedback, as python is not my forte. I definitely could incorporate the `[]` to eliminate the for loop but the issue is that I have to make sure not to throw if the return type is void, which isn't in the `return_type_table` because it means the return statement is just an empty string, eg no `return _some_value_;`

Reply:
Ah, good point.  You can either add `'void' : ''` to `return_type_table`, or use [dict.get](https://docs.python.org/3/library/stdtypes.html#dict.get) to supply a default value and check for the miss after the fact

Reply:
Looking at the code, I now see how to implement it using `[]` and you are 100% right about it being much cleaner. Going to resolve the conversation as we could bikeshed this all day :)